pilot/proxy/envoy/config.go

```diff
@@ -887,19 +896,10 @@ func buildEgressVirtualHost(rule *routing.EgressRule,
 		port.Protocol = model.ProtocolHTTP
 	}
 
-	routes := buildDestinationHTTPRoutes(sidecar, &model.Service{Hostname: destination}, port, instances, config)
+	routes := buildDestinationHTTPRoutesWithDefaultCluster(sidecar, &model.Service{Hostname: destination}, port, instances, config, externalTrafficCluster)
 	// reset the protocol to the original value
 	port.Protocol = protocolToHandle
 
-	if len(routes) > 0 {
-		// Set the destination clusters to the cluster we computed above.
-		// Services defined via egress rules do not have labels and hence no weighted clusters
-		for _, route := range routes {
-			route.Cluster = externalTrafficCluster.Name
-			route.clusters = []*Cluster{externalTrafficCluster}
-		}
```

Comment:
I think you should keep this, and simply `nil` out any weighted cluster in the route. This way, all the custom egress processing is confined inside this function.